sources/dataset.py

```diff
@@ -163,6 +139,73 @@ class (useful for semantic segmentation result printing)
                                      "aggregate": aggregate,
                                      "color": color}
 
+    def save(self, filename):
+        """Save dataset in a json file indicated by `filename`
+
+        Parameter
+        ---------
+        filename: object
+            String designing the relative path where the dataset must be saved
+        """
+        with open(filename, 'w') as fp:
+            json.dump({"image_size": self.image_size,
+                       "classes": self.class_info,
+                       "images": self.image_info}, fp)
+        utils.logger.info("The dataset has been saved into {}".format(filename))
+
+    def load(self, filename, nb_images=None):
+        """Load a dataset from a json file indicated by `filename`
+
+        Parameter
+        ---------
+        filename: object
+            String designing the relative path from where the dataset must be
+        loaded
+        nb_images: integer
+            Number of images that must be loaded (if None, the whole dataset is loaded)
+        """
+        with open(filename) as fp:
+            ds = json.load(fp)
+            self.image_size = ds["image_size"]
+            self.class_info = {int(k):ds["classes"][k] for k in ds["classes"].keys()}
+            if nb_images is None:
+                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys()}
+            else:
+                self.image_info = {int(k):ds["images"][k] for k in ds["images"].keys() if int(k) < nb_images}
+            for img_id, info in self.image_info.items():
+                info['labels'] = {int(k): v for k, v in info['labels'].items()}
+        utils.logger.info("The dataset has been loaded from {}".format(filename))
+
+class MapillaryDataset(Dataset):
+
+    def __init__(self, image_size, glossary_filename):
+        """ Class constructor
+        """
+        Dataset.__init__(self, image_size)
+        self.build_glossary(glossary_filename)
+
+    def build_glossary(self, config_filename):
+        """Read the Mapillary glossary stored as a json file at the data
+        repository root
+
+        Parameter:
```

Comment:
About docstring : add a 's' and whitespace before `:`

Reply:
Done